packages/api/src/getNetwork/getNetwork.ts

```diff
@@ -1,22 +1,25 @@
-import { GEM_WALLET, NetworkResponse, RequestNetworkMessage } from '@gemwallet/constants';
+import { GEM_WALLET, NetworkResponsePayload, RequestGetNetworkMessage } from '@gemwallet/constants';
 
 import { sendMessageToContentScript } from '../helpers/extensionMessaging';
 
-export const getNetwork = async () => {
+export const getNetwork = async (): Promise<NetworkResponsePayload> => {
   /* string: network
    * undefined: something went wrong
    */
-  let response: NetworkResponse | undefined = undefined;
   try {
-    const message: RequestNetworkMessage = {
+    const message: RequestGetNetworkMessage = {
       app: GEM_WALLET,
-      type: 'REQUEST_NETWORK'
+      type: 'REQUEST_GET_NETWORK/V3'
+    };
+    const receivedMessage = await sendMessageToContentScript(message);
+    if (!receivedMessage?.result) {
+      return receivedMessage;
+    }
+    const { network } = receivedMessage?.result;
+    return {
+      result: { network }
     };
-    const { network } = await sendMessageToContentScript(message);
-    response = network;
   } catch (e) {
     throw e;
   }
-
-  return response;
```

Comment:
I think you should keep this or at least return `undefined` just in case the promise doesn't resolve properly on line 14.

Reply:
i have put again the previous logic